plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.test.ts

```diff
@@ -59,94 +55,169 @@ const testFactRetriever: FactRetriever = {
+      },
+      listRegistrations(): FactRetrieverRegistration[] {
+        return [{ factRetriever: retriever, cadence: cron }];
+      },
+      get: (_: string): FactRetrieverRegistration => {
+        return { factRetriever: retriever, cadence: cron };
+      },
+    } as unknown as FactRetrieverRegistry;
+  }
 
-    engine = await FactRetrieverEngine.create(defaultEngineConfig);
+  const databases = TestDatabases.create({
+    ids: ['POSTGRES_13', 'POSTGRES_9', 'SQLITE_3'],
   });
-  it('Should insert facts when scheduled step is run', async () => {
-    (schedule as jest.Mock).mockImplementation(
-      (cronCadence: string, retrieverAction: Function) => {
-        return {
-          cadence: cronCadence,
-          triggerScheduledJobNow: retrieverAction,
-        };
-      },
-    );
 
-    factSchemaAssertionCallback = () => {};
-    factInsertionAssertionCallback = facts => {
-      expect(facts).toHaveLength(1);
-      expect(facts[0]).toEqual({
-        ref: 'test-factretriever',
-        entity: {
-          namespace: 'a',
-          kind: 'a',
-          name: 'a',
-        },
-        facts: {
-          testnumberfact: 1,
+  async function createEngine(
+    databaseId: TestDatabaseId,
+    insert: FactInsertionAssertionCallback,
+    schema: FactSchemaAssertionCallback,
+    cadence?: string,
+    factRetriever?: FactRetriever,
+  ): Promise<FactRetrieverEngine> {
+    const knex = await databases.init(databaseId);
+    const databaseManager: Partial<DatabaseManager> = {
+      forPlugin: (_: string) => ({
+        getClient: async () => knex,
+      }),
+    };
+    const manager = databaseManager as DatabaseManager;
+    const scheduler = new TaskScheduler(manager, getVoidLogger());
+    return await FactRetrieverEngine.create({
+      factRetrieverContext: {
+        logger: getVoidLogger(),
+        config: ConfigReader.fromConfigs([]),
+        discovery: {
+          getBaseUrl: (_: string) => Promise.resolve('http://mock.url'),
+          getExternalBaseUrl: (_: string) => Promise.resolve('http://mock.url'),
         },
+      },
+      factRetrieverRegistry: createMockFactRetrieverRegistry(
+        cadence,
+        factRetriever,
+      ),
+      repository: createMockRepository(insert, schema),
+      scheduler: scheduler.forPlugin('tech-insights'),
+    });
+  }
+
+  it.each(databases.eachSupportedId())(
+    'Should update fact retriever schemas on initialization',
+    async databaseId => {
+      function schemaAssertionCallback(def: FactSchemaDefinition) {
+        expect(def.id).toEqual('test_factretriever');
+        expect(def.version).toEqual('0.0.1');
+        expect(def.entityFilter).toEqual([{ kind: 'component' }]);
+        expect(def.schema).toEqual({
+          testnumberfact: {
+            type: 'integer',
+            description: '',
+          },
+        });
+      }
+
+      engine = await createEngine(
+        databaseId,
+        () => {},
+        schemaAssertionCallback,
+      );
```

Comment:
may wanna add an `expect.assertions(4);` here, to tell jest that you want the test to fail unless all of the four assertions were actually hit in reality too.

Alternatively, you can declare it as a jest fn in the first place

```ts
const schemaAssertionCallback = jest.fn((def: FactSchemaDefinition) => {
 // ... expect stuff
});
```

and then at the bottom, do

```ts
expect(schemaAssertionCallback).toBeCalled();
```

Reply:
Done, thank you!